nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/TicketQuery.java

```diff
@@ -0,0 +1,294 @@
+
+            values.put(valueIndex, value);
+            return this;
+        }
+
+        /**
+         * Validates all provided input for the column, comparator and value. A query is then constructed if the
+         * provided data is valid. A new filter can be built after this method has been called.
+         *
+         * @return this builder.
+         */
+        public Builder completeFilter() {
+            Preconditions.checkState(column != null,
+                    "The column to query must be defined.");
+            Preconditions.checkState(comparator != null,
+                    "The comparator to use in the query must be defined.");
+            Preconditions.checkState(values.size() == comparator.getExpectedValueCount(),
+                    "The comparator expects " + comparator.getExpectedValueCount() + " value(s) but " +
+                            values.size() + " value(s) were provided.");
+
+            switch (comparator) {
+                case BETWEEN:
+                    queries.put(column.getColumnName() + " " + comparator.getComparator() + " ? AND ?", ImmutableMap.copyOf(values)); //Since this will be used to create a prepared statement use a token instead of the value.
+                    break;
+                default:
+                    queries.put(column.getColumnName() + " " + comparator.getComparator() + " ?", ImmutableMap.copyOf(values)); //Since this will be used to create a prepared statement use a token instead of the value.
+                    break;
+            }
+
+            this.column = null;
+            this.comparator = null;
+            values.clear();
+            return this;
+        }
+
+        /**
+         * Simplified method to quickly construct a filter for a ticket query.
+         *
+         * @param column     The {@link Column} to query.
+         * @param comparator The {@link QueryComparator} to use for filtering.
+         * @param values     The value(s) to filter by.
+         * @return this builder.
+         */
+        public Builder filter(Column column, QueryComparator comparator, Object... values) {
+            column(column);
+            comparator(comparator);
+            for (Object value : values) {
+                value(value);
+            }
+            completeFilter();
+
+            return this;
+        }
+
+        /**
+         * Creates a {@link TicketQuery} with the data collected by this builder.
+         *
+         * @return A {@link TicketQuery}.
+         */
+        public TicketQuery build() {
+            return new TicketQuery(this);
+        }
+    }
+
+    /**
+     * An enumeration of columns which can be queried for Ticket related data.
+     */
+    public enum Column {
+        /**
+         * Option to query by a the 'ID' column in the 'Tickets' table. The value must be of the {@link Integer} type.
+         */
+        ID("ID", JDBCType.INTEGER, Integer.class),
```

Comment:
I've updated columns to use another system, if you like it I can update QueryComparator to use a similar one.